presto-parser/src/test/java/io/prestosql/sql/parser/TestSqlParserErrorHandling.java

```diff
@@ -163,6 +163,36 @@ public void testPossibleExponentialBacktracking()
                 "line 1:375: mismatched input '<EOF>'. Expecting: '%', '*', '+', '-', '/', 'AT', 'THEN', '||'");
     }
 
+    @Test
+    public void testPossibleExponentialBacktracking2()
+    {
+        testStatement("SELECT id FROM t WHERE\n" +
+                        "(f()\n" +
+                        "OR (f()\n" +
```

Comment:
```java
repeat("OR (f()\n", 21) +

Reply:
I had it like that because it's easier to copy and use in the CLI, or elsewhere than having to write a program to generate the query programmatically.